test/controllers/categories_controller_test.rb

```diff
@@ -0,0 +1,87 @@
+require "test_helper"
+
+describe CategoriesController do
+  describe "index" do
+    it "succeeds when there are categories" do
+        Category.count.must_equal 3
+        get categories_path
+        must_respond_with :success
+    end
+
+    it "succeeds when there are no categories" do
+      Category.all.each do |item|
+        item.destroy
+      end
+
+      Category.count.must_equal 0
+      get categories_path
+      must_respond_with :success
+    end
+  end
+
+  describe "new" do
+    it "succeeds" do
+      get new_category_path
+      must_respond_with :success
+    end
+  end
+
+  describe "create" do
```

Comment:
You also need to verify that guest users can't create categories, so you're missing some tests.